artman/cli/converter.py

```diff
@@ -0,0 +1,306 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Artman config converter.
+
+It converts the legacy artman config file into the new format. This CLI can be
+removed after all current artman users migrate to the new artman config format.
+"""
+
+from __future__ import absolute_import
+import argparse
+import io
+import json
+import re
+import sys
+import os
+import yaml
+
+
+from collections import OrderedDict
+
+from artman.config.proto.config_pb2 import Artifact, Config
+from google.protobuf.json_format import MessageToJson
+
+
+def main(*args):
+    if not args:
+        args = sys.argv[1:]
+
+    # Get to a normalized set of arguments.
+    try:
+        flags = parse_args(*args)
+        legacy_config = _load_legacy_config_dict(os.path.abspath(flags.config))
+        new_config = _convert(legacy_config)
+        _write_pb_to_yaml(new_config, flags.output)
+    except Exception:
+        print('Fail to convert `%s`.' % flags.config)
+        raise
+
+
+def parse_args(*args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'config',
+        type=str,
+        help='Path to the legacy artman config')
+    parser.add_argument(
+        '--output',
```

Comment:
The tool in its existing form is not careful about what goes to `sys.stdout` vs. `sys.stderr`, which would be a prerequisite to using redirection.